exercises-cpp/mohammed_almakki/ex02_oo_basics/src/ShapeSorter.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include "Shape.h"
+
+class ShapeSorter
+{
+public:
+    ShapeSorter(const std::vector<std::shared_ptr<Shape>> &shapes) : shapes{shapes} {}
+
+    void printShapesWithType(const std::string &type)
+    {
+        for (auto shape : shapes)
```

Comment:
These functions are now at a point where they should no longer be defined inline, but in a separate `.cpp` file. I'm not sure if this is within the scope of the exercise.